### PR TITLE
Renaming `admin` -> `server`

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Doc updates
 - Change `admin.yml` to `dataflow-server.yml`
 - Change package name `admin` to `server`
 - Change shell command `admin` to `dataflow`

Resolves spring-cloud/spring-cloud-dataflow#352

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc [hunk 1]
```diff
@@ -23,31 +23,31 @@ You also need to have link:http://redis.io/[Redis] installed and running if you
 === Deploying 'local'
 
 [start=1]
-1. download the Spring Cloud Data Flow Admin and Shell apps:
+1. download the Spring Cloud Data Flow Server and Shell apps:
 
 ```
-wget http://repo.spring.io/milestone/org/springframework/cloud/spring-cloud-dataflow-admin-local/1.0.0.M2/spring-cloud-dataflow-admin-local-1.0.0.M2.jar
-wget http://repo.spring.io/milestone/org/springframework/cloud/spring-cloud-dataflow-shell/1.0.0.M2/spring-cloud-dataflow-shell-1.0.0.M2.jar
+wget http://repo.spring.io/milestone/org/springframework/cloud/spring-cloud-dataflow-server-local/1.0.0.BUILD-SNAPSHOT/spring-cloud-dataflow-server-local-1.0.0.BUILD-SNAPSHOT.jar
```
Comment (ericbottard) [inline spring-cloud-dataflow-docs/src/main/asciidoc/getting-started.adoc]: Maybe not part of this PR, but it'd be so much better to extract the version in an asciidoc attribute

**Comment by jvalkeal** [comment]
I don't even want to start digging how this impacts server/admin projects. We'll just fix what gets broken as there's really no other way to handle this?

**Comment by markfisher** [comment]
@jvalkeal agreed it's probably easier, if not ideal, to approach this one with "see what breaks"

File: spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc [hunk 2]
```diff
@@ -109,7 +109,7 @@ dependency for local SPI.
 
 ==== Local
 
-1. Open the spring-cloud-data-flow-admin-local/pom.xml in your IDE.
+1. Open the spring-cloud-data-flow-server-local/pom.xml in your IDE.
```
Comment (markfisher) [inline spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc]: looks like we have an extra hyphen here (between data and flow)... was already there, but just noticing it now

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/AdminApplication.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package org.springframework.cloud.dataflow.admin;
+package org.springframework.cloud.dataflow.server;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/AdminApplication.java]: I guess this class will go away once we're done refactoring the various SPI impls, but should we go ahead and rename it in the meantime?

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowServerConfiguration.java [hunk 4]
```diff
@@ -36,13 +36,13 @@
...
 import org.springframework.boot.autoconfigure.web.HttpMessageConverters;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
-import org.springframework.cloud.dataflow.admin.completion.TapOnDestinationRecoveryStrategy;
-import org.springframework.cloud.dataflow.admin.controller.StreamDefinitionController;
-import org.springframework.cloud.dataflow.admin.repository.InMemoryStreamDefinitionRepository;
-import org.springframework.cloud.dataflow.admin.repository.InMemoryTaskDefinitionRepository;
-import org.springframework.cloud.dataflow.admin.repository.StreamDefinitionRepository;
-import org.springframework.cloud.dataflow.admin.repository.TaskDatabaseInitializer;
-import org.springframework.cloud.dataflow.admin.repository.TaskDefinitionRepository;
+import org.springframework.cloud.dataflow.server.completion.TapOnDestinationRecoveryStrategy;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowServerConfiguration.java]: these are no longer in proper alphabetical order after the name change... should re-org imports not just replace the strings

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/FieldValueCounterController.java [hunk 5]
```diff
@@ -1,140 +1,140 @@
-/*
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/FieldValueCounterController.java]: why is this entire class showing as if being replaced in the diff? (formatting?)

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RootController.java [hunk 6]
```diff
@@ -53,20 +53,20 @@
 	private final EntityLinks entityLinks;
 
 	/**
-	 * Construct an {@code AdminController}.
+	 * Construct an {@code ServerController}.
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RootController.java]: this should be `RootController` in the JavaDoc

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDefinitionController.java [hunk 7]
```diff
@@ -24,9 +24,9 @@
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.cloud.dataflow.admin.repository.DuplicateStreamDefinitionException;
-import org.springframework.cloud.dataflow.admin.repository.NoSuchStreamDefinitionException;
-import org.springframework.cloud.dataflow.admin.repository.StreamDefinitionRepository;
+import org.springframework.cloud.dataflow.server.repository.DuplicateStreamDefinitionException;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDefinitionController.java]: need to re-org imports to address alphabetical order

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java [hunk 8]
```diff
@@ -24,8 +24,8 @@
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.cloud.dataflow.admin.repository.NoSuchStreamDefinitionException;
-import org.springframework.cloud.dataflow.admin.repository.StreamDefinitionRepository;
+import org.springframework.cloud.dataflow.server.repository.NoSuchStreamDefinitionException;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java]: re-org imports for alphabetical order

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java [hunk 9]
```diff
@@ -14,12 +14,12 @@
...
  */
 
-package org.springframework.cloud.dataflow.admin.controller;
+package org.springframework.cloud.dataflow.server.controller;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.cloud.dataflow.admin.repository.NoSuchTaskDefinitionException;
-import org.springframework.cloud.dataflow.admin.repository.TaskDefinitionRepository;
+import org.springframework.cloud.dataflow.server.repository.NoSuchTaskDefinitionException;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java]: re-org imports

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDeploymentController.java [hunk 10]
```diff
@@ -22,8 +22,8 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.cloud.dataflow.admin.repository.NoSuchTaskDefinitionException;
-import org.springframework.cloud.dataflow.admin.repository.TaskDefinitionRepository;
+import org.springframework.cloud.dataflow.server.repository.NoSuchTaskDefinitionException;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDeploymentController.java]: re-org imports

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionController.java [hunk 11]
```diff
@@ -14,10 +14,10 @@
  * limitations under the License.
  */
 
-package org.springframework.cloud.dataflow.admin.controller;
+package org.springframework.cloud.dataflow.server.controller;
 
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.cloud.dataflow.admin.repository.NoSuchTaskExecutionException;
+import org.springframework.cloud.dataflow.server.repository.NoSuchTaskExecutionException;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionController.java]: re-org imports

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/TestDependencies.java [hunk 12]
```diff
@@ -14,22 +14,22 @@
...
-import org.springframework.cloud.dataflow.admin.controller.StreamDefinitionController;
-import org.springframework.cloud.dataflow.admin.controller.StreamDeploymentController;
-import org.springframework.cloud.dataflow.admin.controller.TaskDefinitionController;
-import org.springframework.cloud.dataflow.admin.controller.TaskDeploymentController;
-import org.springframework.cloud.dataflow.admin.controller.TaskExecutionController;
-import org.springframework.cloud.dataflow.admin.repository.InMemoryStreamDefinitionRepository;
-import org.springframework.cloud.dataflow.admin.repository.InMemoryTaskDefinitionRepository;
-import org.springframework.cloud.dataflow.admin.repository.StreamDefinitionRepository;
-import org.springframework.cloud.dataflow.admin.repository.TaskDefinitionRepository;
+import org.springframework.cloud.dataflow.server.config.ArtifactRegistryPopulator;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/configuration/TestDependencies.java]: alphabetize imports

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/FieldValueCounterControllerTests.java [hunk 13]
```diff
@@ -1,138 +1,138 @@
-/*
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/FieldValueCounterControllerTests.java]: same weirdness here in the diff... the entire class

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/StreamControllerTests.java [hunk 14]
```diff
@@ -49,9 +49,9 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.test.SpringApplicationConfiguration;
-import org.springframework.cloud.dataflow.admin.configuration.TestDependencies;
-import org.springframework.cloud.dataflow.admin.repository.InMemoryStreamDefinitionRepository;
-import org.springframework.cloud.dataflow.admin.repository.StreamDefinitionRepository;
+import org.springframework.cloud.dataflow.server.configuration.TestDependencies;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/StreamControllerTests.java]: alphabetize imports

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskControllerTests.java [hunk 15]
```diff
@@ -34,9 +34,9 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.test.SpringApplicationConfiguration;
-import org.springframework.cloud.dataflow.admin.configuration.TestDependencies;
-import org.springframework.cloud.dataflow.admin.repository.InMemoryTaskDefinitionRepository;
-import org.springframework.cloud.dataflow.admin.repository.TaskDefinitionRepository;
+import org.springframework.cloud.dataflow.server.configuration.TestDependencies;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskControllerTests.java]: alphabetize imports

File: spring-cloud-dataflow-shell/src/test/java/org/springframework/cloud/dataflow/shell/AbstractShellIntegrationTest.java [hunk 16]
```diff
@@ -28,7 +28,7 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.builder.SpringApplicationBuilder;
 import org.springframework.cloud.dataflow.server.local.LocalDataFlowServer;
-import org.springframework.cloud.dataflow.admin.config.AdminConfiguration;
+import org.springframework.cloud.dataflow.server.config.DataFlowServerConfiguration;
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/test/java/org/springframework/cloud/dataflow/shell/AbstractShellIntegrationTest.java]: alphabetize imports

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/completion/TapOnDestinationRecoveryStrategy.java [hunk 17]
```diff
@@ -14,11 +14,11 @@
  * limitations under the License.
  */
 
-package org.springframework.cloud.dataflow.admin.completion;
+package org.springframework.cloud.dataflow.server.completion;
 
 import java.util.List;
 
-import org.springframework.cloud.dataflow.admin.repository.StreamDefinitionRepository;
+import org.springframework.cloud.dataflow.server.repository.StreamDefinitionRepository;
```
Comment (markfisher) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/completion/TapOnDestinationRecoveryStrategy.java]: alphabetize imports

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/FieldValueCounterController.java (hunk 5 above)
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/FieldValueCounterController.java]: It is weird. not sure why. I just changed the copyright year and package name for this class.

**Comment by markfisher** [comment]
rebased, squashed, and merged as https://github.com/spring-cloud/spring-cloud-dataflow/commit/e74fa829109d5bc883a1d02ed77f36459060a17f
